lib/mayaUsd/sceneIndex/proxyShapeSceneIndexPlugin.cpp

```diff
@@ -56,19 +64,43 @@ HdSceneIndexBaseRefPtr MayaUsdProxyShapeMayaNodeSceneIndexPlugin::_AppendSceneIn
-#if WANT_UFE_BUILD
-    using HdRtidRefDataSource = HdRetainedTypedSampledDataSource<Ufe::Rtid&>;
-    static TfToken         dataSourceRuntimeEntry("runtime");
-    HdDataSourceBaseHandle dataSourceEntryRuntimeHandle = inputArgs->Get(dataSourceRuntimeEntry);
-#endif
+
+    using MayaHydraVersionDataSource = HdRetainedTypedSampledDataSource<int>;
+    static TfToken         dataSourceVersionEntry("version");
+    HdDataSourceBaseHandle dataSourceEntryVersionHandle = inputArgs->Get(dataSourceVersionEntry);
+    int version = 100; // initial mayaHydra version where version data source had not been defined
+    if (auto dataSourceEntryVersion
+        = MayaHydraVersionDataSource::Cast(dataSourceEntryVersionHandle)) {
+        version = dataSourceEntryVersion->GetTypedValue(0.0f);
+    }
 
     if (auto dataSourceEntryNodePath = HdMObjectDataSource::Cast(dataSourceEntryPathHandle)) {
 #if WANT_UFE_BUILD
-        auto dataSourceEntryRuntime = HdRtidRefDataSource::Cast(dataSourceEntryRuntimeHandle);
-        if (TF_VERIFY(dataSourceEntryRuntime, "Error UFE runtime id data source not found")) {
-            Ufe::Rtid& id = dataSourceEntryRuntime->GetTypedValue(0.0f);
-            id = MayaUsd::ufe::getUsdRunTimeId();
-            TF_VERIFY(id != 0, "Invalid UFE runtime id");
+        if (version >= 200) {
+            // Retrieve interpret pick function from the scene index plugin, to be accessed by
+            // mayaHydra interpretRprimPath
+            using MayaHydraInterpretRprimPathDataSource
+                = HdRetainedTypedSampledDataSource<MayaHydraInterpretRprimPath&>;
+            static TfToken         dataSourceInterpretPickEntry("interpretRprimPath");
+            HdDataSourceBaseHandle dataSourceEntryInterpretPickHandle
+                = inputArgs->Get(dataSourceInterpretPickEntry);
+            auto dataSourceEntryInterpretPick
+                = MayaHydraInterpretRprimPathDataSource::Cast(dataSourceEntryInterpretPickHandle);
+            MayaHydraInterpretRprimPath& interpretRprimPath
+                = dataSourceEntryInterpretPick->GetTypedValue(0.0f);
+            interpretRprimPath = (MayaHydraInterpretRprimPath)&MayaUsd::
+                MayaUsdProxyShapeSceneIndex::InterpretRprimPath;
+        } else {
+            using HdRtidRefDataSource = HdRetainedTypedSampledDataSource<Ufe::Rtid&>;
+            static TfToken         dataSourceRuntimeEntry("runtime");
+            HdDataSourceBaseHandle dataSourceEntryRuntimeHandle
+                = inputArgs->Get(dataSourceRuntimeEntry);
+            auto dataSourceEntryRuntime = HdRtidRefDataSource::Cast(dataSourceEntryRuntimeHandle);
+            if (TF_VERIFY(dataSourceEntryRuntime, "Error UFE runtime id data source not found")) {
+                Ufe::Rtid& id = dataSourceEntryRuntime->GetTypedValue(0.0f);
+                id = MayaUsd::ufe::getUsdRunTimeId();
+                TF_VERIFY(id != 0, "Invalid UFE runtime id");
+            }
         }
```

Comment:
What is our policy for backward compatibility?  When can we get rid of this code?

Reply:
I am not sure, but we should at least support 100 which corresponds to the first version of mayaHydra out.